java/client/src/org/openqa/selenium/remote/internal/HttpClientFactory.java

```diff
@@ -114,18 +119,18 @@ public HttpClient getGridHttpClient(int connection_timeout, int socket_timeout)
         .build();
   }
 
-  private SocketConfig createSocketConfig() {
+  private SocketConfig createSocketConfig(int socketTimeout) {
     return SocketConfig.custom()
         .setSoReuseAddress(true)
-        .setSoTimeout(TIMEOUT_THREE_HOURS)
+        .setSoTimeout(socketTimeout > 0 ? socketTimeout : TIMEOUT_THREE_HOURS)
```

Comment:
That check is not comparing against a default value. It is validating the
input and making sure that an invalid value (<= 0) is not set as the
timeout.

On Wed, Apr 1, 2015 at 7:31 PM, Kevin Menard <notifications@github.com>
wrote:

> In
> java/client/src/org/openqa/selenium/remote/internal/HttpClientFactory.java
> <https://github.com/SeleniumHQ/selenium/pull/285#discussion_r27627287>:
>
> >      return SocketConfig.custom()
> >          .setSoReuseAddress(true)
> > -        .setSoTimeout(TIMEOUT_THREE_HOURS)
> > +        .setSoTimeout(socketTimeout > 0 ? socketTimeout : TIMEOUT_THREE_HOURS)
>
> I think rather than checking > 0, it would make things clearer if you had
> a static value like DEFAULT_TIMEOUT = -1 and then use that in
> createHttpClient and change these checks to be socketTimeout !=
> DEFAULT_TIMEOUT. Or something along those lines.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SeleniumHQ/selenium/pull/285/files#r27627287>.
>